tests/mail_test.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+include './src/mail.sh'
+include './tests/utils.sh'
+
+function oneTimeSetUp()
+{
+  declare -gr ORIGINAL_DIR="$PWD"
+  declare -gr FAKE_GIT="$SHUNIT_TMPDIR/fake_git/"
+
+  mk_fake_git "$FAKE_GIT"
+}
+
+function oneTimeTearDown()
+{
+  rm -rf "$FAKE_GIT"
+}
+
+function setUp()
+{
+  declare -gA options_values
+
+  cd "$SHUNIT_TMPDIR" || {
+    ret="$?"
+    fail "($LINENO): Failed to move to shunit temp dir"
+    exit "$ret"
+  }
```

Comment:
It is a bit weird to me that you used `cd` in the setup function. Anyway, for symmetric sake, I think we should use cd to the original path in the `tearDown` function or `cd` inside specific tests.